File: CheckQuota/cq.c
//
// find the user who use storage too much
//
// USAGE : 
// 	./check [MAX USAGE SIZE(MB)]
//

#include<stdio.h>
#include<string.h>

int main(char args, char **argv)
{
	char buffer[255], studentNumber[255];
	short int usage, MAX_USAGE;
	FILE *fp;

	system("du -sm /home/under/* > du.history");
	fp = fopen("./du.history", "r");
	if(!fp) {
		fprintf(stderr, "Error: Can't open history file.\n");
	}	
	
	if(args == 1) {
		MAX_USAGE = 100;
	} else {
		MAX_USAGE = atoi(argv[1]);
	}

	while(fgets(buffer, 255, fp)) {
		if(strlen(buffer) == 0)
			break;
		sscanf(buffer, "%d %s", &usage, studentNumber);
		if(usage > MAX_USAGE) {
			printf("%s : %d MB\n", studentNumber, usage);
		}
	}
	return 0;
}

File: CheckQuota/cq.py
#!/bin/env python

import os
import sys

if __name__=="__main__":
	#os.system("du -sk ./* > du.history")
	file = open("./du.history","r")
	if file == None:
		print "Error: Can't open history file.\n"
		sys.exit(0)
	if len(sys.argv) == 1:
		MAX_USAGE = 100000
	else:
		MAX_USAGE = int(sys.argv[1])
	lines = file.readlines()
	for each in lines:
		(usage, stdNumber) = each.split()
		if int(usage) > MAX_USAGE:
			print '%s : %.d MB' % (stdNumber, int(usage)/1000.0)

